docs/source/compression_preview/config_list.rst

```diff
@@ -0,0 +1,311 @@
+In general, the higher the threshold, the higher the final sparsity. 
+
+.. _global_group_id:
+
+global_group_id
+^^^^^^^^^^^^^^^
+
+``global_group_id`` should jointly used with ``sparse_ratio``.
+All pruning targets that have same ``global_group_id`` will be treat as a whole, and the ``sparse_ratio`` will be distributed across pruning targets.
+That means each pruning target might have different sparse ratio after pruning, but the group sparse ratio will be the configured ``sparse_ratio``.
+
+Note that the ``sparse_ratio`` in the same global group should be the same.
+
+For example, a model has three ``Linear`` modules (``'fc1'``, ``'fc2'``, ``'fc3'``),
+and the expected total sparse ratio of these three modules is 0.5, then the config can be:
+
+.. code-block:: python
+
+    config_list = [{
+        'op_names': ['fc1', 'fc2'],
+        'sparse_ratio': 0.5,
+        'global_group_id': 'linear_group_1'
+    }, {
+        'op_names': ['fc3'],
+        'sparse_ratio': 0.5,
+        'global_group_id': 'linear_group_1'
+    }]
+
+
+.. _dependency_group_id:
+
+dependency_group_id
+^^^^^^^^^^^^^^^^^^^
+
+All pruning targets that have same ``dependency_group_id`` will be treat as a whole, and the positions the targets' pruned will be the same.
+For example, layer A and layer B have same ``dependency_group_id``, and they want to be pruned output channels, then A and B will be pruned the same channel indexes.
+
+Note that the ``sparse_ratio`` in the same dependency group should be the same, and the prunable positions (after reduction by ``granularity``) should be same,
+for example, pruning targets should have same output channel number when pruning output channel.
+
+This key usually be used on modules with add or mul operation, i.e., skip connection.
+
+If you don't know your model structure well, you could use :ref:`auto_set_denpendency_group_ids` to auto detect the dependency operations and auto set their ``dependency_group_id``.
+
+granularity
+^^^^^^^^^^^
+
+Control the granularity of the generated masked.
+
+``default``, ``in_channel``, ``out_channel``, ``per_channel`` and list of integer are supported:
+
+- default: The pruner will auto determine using which kind of granularity, usually consistent with the paper.
+- in_channel: The pruner will do pruning on the weight parameters 1 dimension.
+- out_channel: The pruner will do pruning on the weight parameters 0 dimension.
+- per_channel: The pruner will do pruning on the input/output -1 dimension.
+- list of integer: Block sparse will be applied. For example, ``[4, 4]`` will apply 4x4 block sparse on the last two dimensions of the weight parameters.
+
+Note that ``in_channel`` or ``out_channel`` is not supported for input/output targets, please using ``per_channel`` instead.
+``torch.nn.Embedding`` is special, it's output dimension on weight is 1, so if want to pruning Embedding output channel, please set ``in_channel`` for its granularity for workaround.
```

Comment:
why is there a special case for embedding?

Reply:
because most layer output related to weight dim 0, but embedding is related to weight dim 1.